src/main/java/bio/terra/cli/command/resource/create/BqDataset.java

```diff
@@ -20,7 +20,7 @@
   @CommandLine.Mixin ControlledResourceCreation controlledResourceCreationOptions;
   @CommandLine.Mixin BqDatasetLifetime bqDatasetLifetimeOptions;
 
-  @CommandLine.Option(names = "--dataset-id", required = true, description = "BigQuery dataset id.")
+  @CommandLine.Option(names = "--dataset-id", description = "BigQuery dataset id.")
```

Comment:
nit: include a note here about how we will generate a name if not provided, similar to the `create gcp-notebook` command. ditto other command classes